cli/src/cli_util.rs

```diff
@@ -3701,6 +3701,25 @@ impl CliRunner {
             process_global_args_fn(ui, &matches)?;
         }
 
+        config_env.set_command(
+            if let Some((subcommand, mut subcommand_matches)) = matches.subcommand() {
+                let mut subcommand_combined = String::from(subcommand);
+                while let Some((subcommand, new_matches)) = subcommand_matches.subcommand() {
+                    subcommand_combined.push(' ');
+                    subcommand_combined.push_str(subcommand);
+                    subcommand_matches = new_matches;
+                }
+                config = config_env.resolve_config(&raw_config)?;
+                migrate_config(&mut config)?;
+                ui.reset(&config)?;
```

Comment:
nit: deduplicate this. Perhaps, `ui.reset()` can be omitted as we do that later.